### PR TITLE
[feat] Move the updater code from the launcher

### PR DESCRIPTION
**Pull request by meskio**
The new pyinstaller method will rule out the launcher, we need the
updater working inside bitmask client.

- Closes: #7291

### CODE REVIEW

File: src/leap/bitmask/updater.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# backend_app.py
```
Comment (ivanalejandro0) [inline src/leap/bitmask/updater.py]: updater.py

**Comment by ivanalejandro0** [comment]
Maybe you could add some docstrings to the methods on the updater.

File: src/leap/bitmask/updater.py (hunk 1 above)
Comment (meskio) [inline src/leap/bitmask/updater.py]: ups copy-paste error :D

File: src/leap/bitmask/backend_app.py [hunk 2]
```diff
@@ -80,23 +82,21 @@ def run_backend(bypass_checks=False, flags_dict=None, frontend_pid=None):
...
     backend = LeapBackend(bypass_checks=bypass_checks,
                           frontend_pid=frontend_pid)
     backend.run()
 
 
+def start_events_and_updater():
+    event_server.ensure_server()
+
+    if flags.STANDALONE:
+        from leap.bitmask.updater import Updater
```
Comment (kalikaneko) [inline src/leap/bitmask/backend_app.py]: can you do this import inside a conditional, and just log if it cannot be imported?
my plans for the debian package is to remove the updater module. thoughts?

File: src/leap/bitmask/updater.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+# Copyright (C) 2014, 2015 LEAP
```
Comment (kalikaneko) [inline src/leap/bitmask/updater.py]: this is only 2015 :P

Comment (meskio) [inline src/leap/bitmask/updater.py]: But the code was written in 2014 in another repo.

File: src/leap/bitmask/backend_app.py (hunk 2 above)
Comment (meskio) [inline src/leap/bitmask/backend_app.py]: You mean to do a try/catch? I already do the import inside a condition thinking on that, but the try/catch would be a good addition.

Comment (kalikaneko) [inline src/leap/bitmask/backend_app.py]: yes, try. my thinking is that the flag can be also added to debian package, and we'd be better just by not including that code.

File: src/leap/bitmask/updater.py (hunk 3 above)
Comment (kalikaneko) [inline src/leap/bitmask/updater.py]: aah, makes sense then XD

File: src/leap/bitmask/backend_app.py [hunk 4]
```diff
@@ -80,23 +82,25 @@ def run_backend(bypass_checks=False, flags_dict=None, frontend_pid=None):
...
 
+def start_events_and_updater(logger):
+    event_server.ensure_server()
+
+    if flags.STANDALONE:
+        try:
+            from leap.bitmask.updater import Updater
+            updater = Updater()
+            updater.start()
+        except Exception as e:
```
Comment (kalikaneko) [inline src/leap/bitmask/backend_app.py]: maybe catch ImportError explicitely, and log that updates are not enabled in this distribution?

**Comment by ivanalejandro0** [comment]
rebased & merged.

**Pull request closed without merging**